PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/DiningHallWidgetAdapter.kt

```diff
@@ -0,0 +1,156 @@
+
+        // Connection to data source
+        override fun onCreate() {
+            //connect to data source
+            mDiningRequest = DiningHallWidget.diningRequestInstance
+            getWidgetDiningHalls()
+        }
+
+        // The place where we fetch data from the source set new data and update collection widget accordingly
+        override fun onDataSetChanged() {
+            //refresh data -> Update data every 30 minutes
+        }
+
+        override fun onDestroy() {
+            //close connection to data source
+        }
+
+        override fun getCount(): Int {
+            Log.d("msg", "${dataSet.size}")
+            return dataSet.size
+        }
+
+        override fun getViewAt(position: Int): RemoteViews {
+            val views = RemoteViews(context.packageName, R.layout.dining_hall_widget_item)
+            Log.d("msg", "position: ${position}, name: ${dataSet[position].name}")
+            views.setTextViewText(R.id.appwidget_text, dataSet[position].name)
+            views.setImageViewResource(R.id.diningBackground, dataSet[position].image)
+            if (dataSet[position].isOpen) {
+                Log.d("msg", "Open right now")
+                views.setImageViewResource(R.id.imageView, R.drawable.baseline_check_circle_24)
+                views.setInt(R.id.textView3, "setBackgroundColor", Color.parseColor("#6DB786"))
+                if (dataSet[position].openMeal() != "all" && dataSet[position].openMeal() != null) {
+                    val resources = context.resources
+                    val open_label : String = resources.getString(getOpenStatusLabel(dataSet[position].openMeal() ?: ""))
+                    views.setTextViewText(R.id.textView3, open_label)
+                }
+            }
+            else {
+                views.setImageViewResource(R.id.imageView, R.drawable.baseline_cancel_24)
+                views.setInt(R.id.textView3, "setBackgroundColor", Color.parseColor("#990000"))
+                if (dataSet[position].openTimes().isEmpty()) {
+                    views.setTextViewText(R.id.textView3, "Closed Today")
+                }
+                else {
+                    views.setTextViewText(R.id.textView3, "Closed")
+                }
+            }
+            val extras = Bundle()
+            extras.putInt("key_data", position)
+            val fillInIntent = Intent()
+            fillInIntent.putExtras(extras)
+            views.setOnClickFillInIntent(R.id.widgetBackground, fillInIntent)
+            return views
+        }
+
+        override fun getLoadingView(): RemoteViews? {
+            return null
+        }
+
+        override fun getViewTypeCount(): Int {
+            return 1
+        }
+
+        override fun getItemId(position: Int): Long {
+            return position.toLong()
+        }
+
+        override fun hasStableIds(): Boolean {
+            return true
+        }
+
+        @SuppressLint("CheckResult")
```

Comment:
I don't think we need this annotation anymore, now that we are using retrofit instead of retrofit2.